### ISSUE TITLE
find_between used by topology controller to validate a topology being add assumes OXP URL has a .net domain

### ISSUE THREAD

**Issue by italovalcy** [atlanticwave-sdx/sdx-lc#108]
Hi,

The topology_controller `add_topology()` API handler run a validation filter on the payload received to validate the topology `id` matches the SDXLC_DOMAIN: https://github.com/atlanticwave-sdx/sdx-lc/blob/main/swagger_server/controllers/topology_controller.py#L56-L57

This strategy can lead to filtering out OXP's which URL does not have the `.net` on the domain name, such as `tenet.ac.za`. In my opinion, we should not assume all OXPs will have the `.net` domain.

that being said, maybe we should change a bit the validation there to be more open to accept other domains.

**Comment by sajith** [comment]
Agree with Italo that we should accomodate TLDs that are not `.net` also. We could probably could use something like:

```python
# message_id could be in the form of "urn:ogf:network:sdx:topology:zaoxi.net".
message_id = body.get("id")

if message_id:
    *_, domain_name = message_id.split(":")
else:
   return "Message ID is missing or invalid", 400
```

**Comment by sajith** [comment]
Or maybe we can use:

```python
if not message_id.endswith(SDXLC_DOMAIN):
    return "Domain name in topology does not match LC domain", 400
```

So `SDXLC_DOMAIN` now will have to contain the second level domain and TLD (as in: `zaoxi.net`, not just `zaoxi`).  Hopefully we do not use the same assumption anywhere else.

**Comment by congwang09** [comment]
> Or maybe we can use:
> 
> ```python
> if not message_id.endswith(SDXLC_DOMAIN):
>     return "Domain name in topology does not match LC domain", 400
> ```
> 
> So `SDXLC_DOMAIN` now will have to contain the second level domain and TLD (as in: `zaoxi.net`, not just `zaoxi`). Hopefully we do not use the same assumption anywhere else.

I think this one can be better. Then we will need some agreement with OXP operators. @italovalcy what would be the right SDX LC domain name here? Should we include the `.net` or `.ac.za`?

If we don't want the part `.net`, we can use `.` instead of `.net` to find between.

**Comment by italovalcy** [comment]
Hi team, I believe this issue was fixed by PR  #114, right? The idea on PR  #114 is very aligned with what we discussed on the integration meeting. Thus, I think we are good with the solution there. I'm closing this issue as resolved, but please feel free to re-open it
